src/providers/tfc/runProvider.ts

```diff
@@ -21,10 +49,127 @@ export class RunTreeDataProvider implements vscode.TreeDataProvider<vscode.TreeI
   }
 
   getChildren(element?: vscode.TreeItem | undefined): vscode.ProviderResult<vscode.TreeItem[]> {
-    return element ? [element] : [];
+    if (element) {
+      return [element];
+    }
+    if (!this.activeWorkspace) {
+      return [];
+    }
+
+    try {
+      return this.getRuns(this.activeWorkspace.id, this.activeWorkspace.name);
+    } catch (error) {
+      return [];
+    }
+  }
+
+  private async getRuns(workspaceId: string, workspaceName: string): Promise<vscode.TreeItem[]> {
+    const organization = this.ctx.workspaceState.get('terraform.cloud.organization', '');
+    if (organization === '') {
+      return [];
+    }
+
+    const session = await vscode.authentication.getSession(TerraformCloudAuthenticationProvider.providerID, [], {
+      createIfNone: false,
+    });
+
+    if (session === undefined) {
+      return [];
+    }
+
+    try {
+      const runs = await apiClient.listRuns({
+        params: { workspace_id: workspaceId },
+      });
+
+      if (runs.data.length === 0) {
+        return [{ label: `No runs found for ${workspaceName}` }];
+      }
+
+      const items: RunTreeItem[] = [];
+      for (let index = 0; index < runs.data.length; index++) {
+        const run = runs.data[index];
+        items.push(new RunTreeItem(run.id, workspaceName, run.attributes));
+      }
+
+      // TODO: pagination
+
+      return items;
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response?.status === 401) {
+        vscode.window.showErrorMessage(`Runs not accessible for workspace ${workspaceId}: ${error}`);
+      } else if (error instanceof Error) {
+        vscode.window.showErrorMessage(error.message);
+      } else if (typeof error === 'string') {
+        vscode.window.showErrorMessage(error);
+      }
+      return [];
+    }
   }
 
   dispose() {
     //
   }
 }
+
+export class RunTreeItem extends vscode.TreeItem {
+  constructor(public id: string, public workspaceName: string, public attributes: RunAttributes) {
+    super(attributes.message, vscode.TreeItemCollapsibleState.None);
+    this.id = id;
+
+    this.workspaceName = workspaceName;
+    this.iconPath = this.getStatusIcon(attributes.status);
+    this.description = `${attributes['trigger-reason']} ${attributes['created-at']}`;
+    this.tooltip = new vscode.MarkdownString(`
+### ID: \`${id}\`
+
+${attributes.message}
+
+ - **Status**: ${attributes.status}
+ - **Trigger reason**: ${attributes['trigger-reason']}
+ - **Created at**: ${attributes['created-at']}
+ - **Source**: ${attributes.source}
+ - **Terraform** ${attributes['terraform-version']}
```

Comment:
```suggestion
### \`${workspaceName}\`
#### ID: \`${id}\`
___
${attributes.message}
___
| | |
--|--
| **Status**               | ${attributes.status}|
| **Trigger reason** | ${attributes['trigger-reason']}|
| **Created at**        | ${attributes['created-at']}|
| **Source**              | ${attributes.source}|
| **Terraform**         | ${attributes['terraform-version']}|
```

Love what is there already. This suggestion brings it a little closer to what the TFC page looks like while also spacing things automatically